training/controllers/auth.py

```diff
@@ -21,17 +22,26 @@ def user_to_g():
         user = db.session.query(User).filter_by(id=username).first()
         g.user = user
     elif header is not None:
-        username_token = serializer.loads(header, salt='login')
+        try:
+            username_token = serializer.loads(header, salt='login')
+        except BadSignature:
+            username_token = None
 
         user = db.session.query(User).filter_by(id=username_token).first()
-        g.user = user
+
+        # We check if user is None, because we could pass a wrong auth_token through the api
+        if user is not None:
+            g.user = user
+        else:
+            g.user = None
     else:
         g.user = None
     return None
 
 
 def test_unique_fields(form):
-    if User.query.filter_by(id=form['username']).first() or User.query.filter_by(email=form['email']).first() is not None:
+    if User.query.filter_by(id=form['username']).first() or User.query.filter_by(
+            email=form['email']).first() is not None:
```

Comment:
Please prevent code repetition. Here you have almost the same verification as in `api.py:12` but the email check us using `email` in the `find_by()`

Reply:
You are right, i created a functions called test_unique_fields that will be imported in both modules to solve that auth task.